stock_analysis/models/stock_analysis.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# © 2016 Lorenzo Battistini - Agile Business Group - I.A.S. Ingenieria, Aplicaciones y Software
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import tools
+from openerp import models, fields
+
+
+class StockAnalysis(models.Model):
+    _name = 'stock.analysis'
+    _auto = False
+    _rec_name = 'product_id'
+
+    product_id = fields.Many2one(
+        'product.product', string='Product', readonly=True)
+    location_id = fields.Many2one(
+        'stock.location', string='Location', readonly=True)
+    qty = fields.Float(string='Quantity', readonly=True)
+    lot_id = fields.Many2one(
+        'stock.production.lot', string='Lot', readonly=True)
+    package_id = fields.Many2one(
+        'stock.quant.package', string='Package', readonly=True)
+    in_date = fields.Datetime('Incoming Date', readonly=True)
+    categ_id = fields.Many2one(
+        'product.category', string='Category', readonly=True)
+    company_id = fields.Many2one(
+        'res.company', string='Company', readonly=True)
+
+    def init(self, cr):
```

Comment:
You can use the same technique as Odoo standard modules that split the query in several parts and call a method for getting that part. This way, you easy a lot the inheratibility of this module. See an example here: https://github.com/odoo/odoo/blob/e318a964901b90eabfbb970a0a2a756bc3e3d7fa/addons/account/report/account_invoice_report.py#L161-L177